### PR TITLE
Sync course subject names from ttapi [3/3]

### PR DESCRIPTION
**Pull request by despo**
## Context

Removes `subject_codes` field from `courses` table

## Link to Trello card

https://trello.com/c/81C0pJrz/3583-sync-course-subject-names-from-ttapi-3

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by mnacos — COMMENTED** [review]
Looks very good. I would approve if it wasn't a draft! It even covers exports via the `.subject_codes` helper method on course. This could return an empty list, of course, if a sync hasn't happened, but maybe we could trigger one as part of the deploy?

**Comment by despo** [comment]
@mnacos As syncs happen hourly I expect one to have been done prior to this being merged (given it has already worked in QA and Staging, and part 2 still needs to be merged), so I don't think we need any manual intervention! 

I'm waiting for part 2 to be approved, merged and deployed though before opening the PR for review just to avoid any confusion or accidental merge!

**Comment by SamJamCul** [comment]
Should we do the migration in a separate PR?

**Comment by despo** [comment]
@SamJamCul every other dependency has already been satisfied through PRs #4489 and #4490

**Comment by SamJamCul** [comment]
Would somewhere like [this](https://github.com/DFE-Digital/apply-for-teacher-training/blob/7ab6bdff99c91efe7e6b179401b192fbcfd73a62/app/services/data_api/tad_application_export.rb#L62) also need to have the `subject_codes` removed?

**Comment by despo** [comment]
@SamJamCul No as I have introduced a method as part of https://github.com/DFE-Digital/apply-for-teacher-training/pull/4490 that maps subject.code to `subject_codes` for simplicity.

File: spec/system/find_sync/syncing_providers_spec.rb
Comment (cwrw) [inline spec/system/find_sync/syncing_providers_spec.rb:48]: can we not keep this as `subject_codes` as we defined the helper method to map those?

**Review by cwrw — APPROVED** [review]
Looks good, just the one comment

File: spec/system/find_sync/syncing_providers_spec.rb
Comment (despo) [reply spec/system/find_sync/syncing_providers_spec.rb:48]: We could, however this reflects what the sync actually does and it doesn't require digging into the course code in order to understand why subjects is set and subject_codes is returned.

**Pull request merged**
